cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedStreamSizeScheduleStore.java

```diff
@@ -311,15 +334,33 @@ public void apply() throws Exception {
           if (txMethod != null) {
             txMethod.execute();
           }
-          byte[] rowKey = Bytes.toBytes(String.format("%s:%s", KEY_PREFIX,
-                                                      AbstractSchedulerService.scheduleIdFor(programId, programType,
-                                                                                             scheduleName)));
+          byte[] rowKey = Bytes.toBytes(getRowKey(programId, programType, scheduleName));
```

Comment:
Check for the existence of new format when the background thread runs, is in a separate PR. In that PR, when the background thread runs, it will check if the data exists in the new format, in which case, it will just delete the old format info and won't change the new format.